docs/design/architecture.md

```diff
@@ -213,6 +213,75 @@ this will generally start off `true` and then become `false` as the segment is p
 published segments. Generally this is a transient state, and segments in this state will soon have their `used` flag
 automatically set to false.
 
+### Availability and consistency
+
+Druid has an architectural separation between ingestion and querying, as described above in
+[Indexing and handoff](#indexing-and-handoff). This means that when understanding Druid's availability and
+consistency properties, we must look at each function separately.
+
+On the **ingestion side**, Druid's primary [ingestion methods](../ingestion/index.md#ingestion-methods) are all
+pull-based and offer transactional guarantees. This means that you are guaranteed that ingestion using these will
+publish in an all-or-nothing manner:
+
+- Supervised "seekable-stream" ingestion methods like [Kafka](../development/extensions-core/kafka-ingestion.md) and
+[Kinesis](../development/extensions-core/kinesis-ingestion.md). With these methods, Druid commits stream offsets to its
+[metadata store](#metadata-storage) alongside segment metadata, in the same transaction. Note that ingestion of data
+that has not yet been published can be rolled back if ingestion tasks fail. In this case, partially-ingested data
+discarded, and Druid will resume ingestion from the last committed set of stream offsets. This ensures exactly-once
```

Comment:
partially-ingested data discarded -> partially-ingested data "is" discarded

Reply:
Thanks, fixed.